### PR TITLE
Update build an element tutorial

### PR DESCRIPTION
**Pull request by katejeffreys**

### CODE REVIEW

**Comment by arthurevans** [comment]
Need to add this to nav.yaml.

File: app/2.0/start/first-element/intro.md [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+-   [git](https://git-scm.com/downloads).
+-   The starting code, [available on GitHub](https://github.com/PolymerLabs/polymer-2-first-element.git).
+-   The [Polymer CLI](/2.0/docs/tools/polymer-cli) to run the demo.
+
+### Fork the starting code repo
+
+1.  Fork the repo by running this command:
+
+    ```bash
+    git clone https://github.com/PolymerLabs/polymer-2-first-element.git
```
Comment (arthurevans) [inline app/2.0/start/first-element/intro.md]: We could save some time (like we did in 1.x) by supplying a downloadable file with the source code+dependencies pre-installed.

File: app/2.0/start/first-element/step-3.md [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+
+demo-element.html—existing demo code { .caption }
+
+```html
+<icon-toggle toggle-icon="star" pressed></icon-toggle>
+```
+
+These lines are _already in place_, you don't have to add them. These lines
+are part of the demo you see on screen. But if you'd like to experiment, try
+adding a new `<icon-toggle>` element to the `icon-toggle-demo.html` file. Some
```
Comment (arthurevans) [inline app/2.0/start/first-element/step-3.md]: demo-element.html

File: app/2.0/start/first-element/step-3.md [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+icon names you can try are `add`, `menu`, and `settings`.
+
+**Learn more: attribute and property names.** You'll note that the markup above
+uses `toggle-icon`, not `toggleIcon`. Polymer represents camelCase property names
+using dash-case attribute names. To learn more, see <a href="/2.0/docs/devguide/properties#property-name-mapping">Property
+name to attribute name mapping</a> in the Polymer library docs.
+{ .alert .alert-info }
+
+Next, add a declaration for the `toggleIcon` property.
```
Comment (arthurevans) [inline app/2.0/start/first-element/step-3.md]: Note: in 2.0, this step is going to need to move up. 2.0 requires the property to be declared in the `properties` object if you want to be able to configure it from markup.

File: app/2.0/start/first-element/step-4.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
+---
+title: "Step 4: React to input"
+subtitle: "Build your first Polymer element"
+---
+
+Of course, a button isn't a button if you can't click it. 
+
+To toggle the button, we will add an event listener. To make event listener methods available, we will include the methods we need by importing and using the `Polymer.GestureEventListeners` mixin.
```
Comment (arthurevans) [inline app/2.0/start/first-element/step-4.md]: 2nd sentence here is wrong. The mixin adds support for gesture events that work the same way across different browsers. (You should also probably introduce the concept of mixins somewhere in this step.)

At some point, you might say something like, GestureEventListeners provides a `tap` event that works like the standard `click` event on desktop browsers, and like a native tap event on mobile browsers.

Maybe @azakus could tell us whether this is really still necessary...

**Comment by arthurevans** [comment]
A few comments for today, will try to review more thoroughly tomorrow.

File: app/2.0/start/first-element/step-4.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+*   The `tap` event is generated by Polymer's [gesture system](/2.0/docs/devguide/gesture-events)
+    when the user clicks or taps on a target with a mouse or finger.
+
+Save the `icon-toggle.html` file and look at the demo again. You should be able to press the button and see it
+toggle between its pressed and unpressed states.
+
+<img src="/images/2.0/first-element/databound-toggles.png" alt="Demo showing icon toggles with star and heart icons.">
+
+**Learn more: data binding.** To see how the demo works, open `icon-toggle-demo.html`
```
Comment (arthurevans) [inline app/2.0/start/first-element/step-4.md]: `demo-element.html`

File: app/2.0/start/first-element/step-5.md [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+Frequently you want to define custom property values at the document level, to
+set a theme for an entire application, for example. Because custom properties
+aren't built into most browsers yet, you need to use a special `custom-style`
+tag to define custom properties outside of a Polymer element. Try
+adding the following code inside the `<head>` tag of your `index.html` file:
+
+```
+<custom-style>
+  <style>
+    /* Define a document-wide default—will not override a :host rule in icon-toggle-demo */
```
Comment (arthurevans) [inline app/2.0/start/first-element/step-5.md]: search & replace icon-toggle-demo => demo-element

File: app/2.0/start/first-element/intro.md [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+    If you’re wondering why `polymer serve` does this, see [HTML imports and dependency
+    management](/2.0/docs/tools/polymer-cli#element-project-layout).)
+
+    You’ll see some text where the icon toggles should appear. It doesn’t look
+    very interesting, but it shows that everything is working.
+
+<img src="/images/2.0/first-element/starting-state.png" alt="Initial state of the demo. The demo shows three icon-toggle elements, two labeled 'statically-configured icon toggles' and one labeled 'data-bound icon toggle'. Since the icon toggles are not implemented yet, they appear as placeholder text reading 'Not much here yet'." title="Initial demo">
+
+**If this text doesn’t appear**, make sure you’re looking at the the demo folder itself, or at `demo/index.html`,
+not at `toggle-icon.html` or `toggle-icon-demo.html`.
```
Comment (arthurevans) [inline app/2.0/start/first-element/intro.md]: `toggle-icon-demo.html` => `demo-element.html`

**Comment by arthurevans** [comment]
(Oh, also there's a merge conflict, but it should be pretty straightforward.

**Comment by wmginsberg** [comment]
working on a few changes, but the main one is the dependencies in bower.json are still pointing at 1.0.0. Should be (and updated again when we release):
 "dependencies": {
    "polymer": "Polymer/polymer#2",
    "iron-icon": "PolymerElements/iron-icon#2.0-preview",
    "iron-icons": "PolymerElements/iron-icons#2.0-preview"
  } 

Doc with rest of changes here: 
https://docs.google.com/document/d/1Th7n88SHiR3BhLadqkJ1T2YEIsioR9jy6OZAy1NX6AI/edit?hl=en

**Comment by katejeffreys** [comment]
@Wendy, thanks so much for the review! I have commented on the doc to
request some clarifications.

Kate

File: app/2.0/start/first-element/intro.md [hunk 8]
```diff
@@ -58,23 +59,21 @@ To follow this tutorial, you’ll need:
...
 
-2.  Install the latest version of [Bower](http://bower.io/#install-bower).
-
-        npm install -g bower
-
-3.  Install Polymer CLI.
-
-        npm install -g polymer-cli
+   ```bash
+   npm install -g polymer-cli@next
```
Comment (abdonrd) [inline app/2.0/start/first-element/intro.md]: I think we can remove the `@next`. :)

Comment (katejeffreys) [inline app/2.0/start/first-element/intro.md]: Thanks!

**Pull request merged**